src/NuGet.Clients/PackageManagement.UI/Xamls/InfiniteScrollList.xaml.cs

```diff
@@ -271,7 +267,10 @@ private void HandleItemLoaderStateChange(IItemLoader<PackageItemListViewModel> l
 
                     if (!Items.Contains(_loadingStatusIndicator))
                     {
-                        Items.Add(_loadingStatusIndicator);
+                        lock (_itemsLock)
```

Comment:
There will always be single thread adding actual data and in some cases, could be another thread adding loading status. So I doubt there will ever be a case where multiple threads are adding same stuff, which is why i haven't put contains or other read only properties under lock.